experiments/mine_dataflow.py

```
@@ -0,0 +1,230 @@
+    print(f"file: {path}")
+    print(f"instructions: {total:,}  bytes: {total_bytes:,}")
+    print()
+    print("=== Provable in-block redundancies ===")
+    for k in ("zext_pairs", "zext_pairs_redundant", "zext_pairs_redundant_bytes",
+              "sext_ops", "sext_ops_redundant", "sext_ops_redundant_bytes",
+              "sp_loads", "sp_loads_value_already_in_reg",
+              "sp_loads_value_already_in_reg_bytes", "sp_loads_same_reg_noop",
+              "sp_stores"):
+        print(f"  {k:<42} {res[k]:>12,}")
+    print()
+    print(f"=== Offset relaxation ===")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)